crypto/x509/x509_cmp.c

```diff
@@ -163,6 +163,14 @@ int X509_cmp(const X509 *a, const X509 *b)
     return rv;
 }
 
+int X509_cmp_ex(X509 *a, X509 *b,
+                OPENSSL_CTX *libctx, const char *propq)
+{
+    X509_set0_libctx(a, libctx, propq);
+    X509_set0_libctx(b, libctx, propq);
```

Comment:
Surely the whole point of adding the libctx to the x509 structure is that we don't have to change all the `X509_*` calls to additionally take a libctx. So, I'm not sure why this one gets singled out for special treatment.

Reply:
Fixed